awx/main/wsrelay.py

```diff
@@ -209,51 +209,55 @@ def __init__(self):
+                    if logger.isEnabledFor(logging.DEBUG):
+                        logmsg = "{} {}".format(logmsg, payload)
+                        logger.warning(logmsg)
                     return
-                self.known_hosts[hostname] = ip
-                logger.debug(f"Web host {hostname} ({ip}) online heartbeat received.")
-            elif payload.get("action") == "offline":
-                hostname = payload.get("hostname")
-                ip = payload.get("ip")
-                if ip is None:
-                    # If we don't get an IP, just try the hostname, maybe it resolves
-                    ip = hostname
-                if ip is None:
-                    logger.warning(f"Received invalid offline ws_heartbeat, missing hostname and ip: {payload}")
+
+                # Skip if the message comes from the same host we are running on
+                # In this case, we'll be sharing a redis, no need to relay.
+                if payload.get("hostname") == self.local_hostname:
                     return
-                self.cleanup_offline_host(ip)
-                logger.debug(f"Web host {hostname} ({ip}) offline heartbeat received.")
-        except Exception as e:
-            # This catch-all is the same as the one above. asyncio will eat the exception
-            # but we want to know about it.
-            logger.exception(f"on_ws_heartbeat exception: {e}")
+
+                if payload.get("action") == "online":
+                    hostname = payload.get("hostname")
+                    ip = payload.get("ip")
+                    if ip is None:
+                        # If we don't get an IP, just try the hostname, maybe it resolves
+                        ip = hostname
+                    if ip is None:
+                        logger.warning(f"Received invalid online ws_heartbeat, missing hostname and ip: {payload}")
+                        return
+                    self.known_hosts[hostname] = ip
+                    logger.debug(f"Web host {hostname} ({ip}) online heartbeat received.")
+                elif payload.get("action") == "offline":
+                    hostname = payload.get("hostname")
+                    ip = payload.get("ip")
+                    if ip is None:
+                        # If we don't get an IP, just try the hostname, maybe it resolves
+                        ip = hostname
+                    if ip is None:
+                        logger.warning(f"Received invalid offline ws_heartbeat, missing hostname and ip: {payload}")
+                        return
+                    self.cleanup_offline_host(ip)
+                    logger.debug(f"Web host {hostname} ({ip}) offline heartbeat received.")
+            except Exception as e:
+                # This catch-all is the same as the one above. asyncio will eat the exception
+                # but we want to know about it.
+                logger.exception(f"on_ws_heartbeat exception: {e}")
```

Comment:
_requests review from @relrod_

Reply:
Yeah, I wrote this patch with @john-westcott-iv a few weeks ago. It should be reasonable. (If you view the github diff and tell it to ignore whitespace, it gives a clearer picture of the changes in that file.)

:+1: here (on wsrelay -- I haven't reviewed the rest)